cmd/subnetcmd/upgradecmd/validate.go

```diff
@@ -55,9 +57,19 @@ func validateUpgradeBytes(file []byte, lockFile []byte) ([]params.PrecompileUpgr
 		return nil, err
 	}
 
-	for _, ts := range allTimestamps {
-		if time.Unix(ts, 0).Before(time.Now()) {
-			return nil, errBlockTimestampInthePast
+	if !force {
```

Comment:
So you're not actually accepting force as an argument. It's not a registered flag in the apply command. You should probably pass it an as argument to this function instead of reading from the global.

Reply:
Also, I didn't catch this earlier, but I don't love this file. Every other file in the `cmd` directory tree is either its own command or a test. The code in this file should either move to its own package or be part of `apply.go`

Reply:
That's a self-chosen pattern though, which to me is peculiar (I don't see why instead of cluttering big files, one should not be able to separate logic into different, easier to maintainable files. In fact I'd argue that's way better of a pattern, which allows files to be of manageable size).

But ok, I will adhere to your request.